Lib/fontTools/misc/py23.py

```diff
@@ -71,7 +71,7 @@ def strjoin(iterable, joiner=""):
 
 
 def tobytes(s, encoding="ascii", errors="strict"):
```

Comment:
i'd expected a method called `tobytes` to always return a `bytes` object, not sometimes a bytes other times a bytearray. Maybe you should add another `if isinstance(s, bytearray): bytes(s)`

Reply:
I generally agree. But I don't see the point of an extra copy.  If we were to do that though, we can just always make the default path return `bytes(s)`, which like Tuple type, is a no-op if input is bytes already.

So, my changes in this branch make some internal data types to be left as bytearray instead of bytes after, eg. subsetting. I'm treating the two types interchangable for readonly use purposes. I feel like that's how they were designed to be.

And I removed tobytes() invocations where I could audit that we never get a str there. I think my branch doesn't depend on this change per se. I made it because it looks correct; since the `.encode()` method is a property of `str` only. There has been an assumption that what's passed to this function is either `bytes` or `str`. And under that assumption this change is still correct.

Reply:
Okay, I made it always return bytes.

Reply:
> I'm treating the two types interchangable for readonly use purposes.

that's ok.

> I made it because it looks correct; since the .encode() method is a property of str only. 

true. Then for the same logic, we should change tounicode as well..

> make the default path return bytes(s)

yeah I feel that would be better, but i'm concerned for the extra copy.
Imagine later we decide to add type hints to these functions, what would the type of the return value be? Depends on the type of the input. I guess with `typing.overload` one could describe that. Though it's just weid that if one passes a ramdon object that isn't `str` to tobytes the latter returns it back as it was without complaining.

Reply:
I've made it return `bytes(s)` for now.

If typing, we should make it a generic that passes bytes-like objects intact, where bytes-like is defined as implementing the buffer protocol with item type of unsigned byte.

Reply:
> Then for the same logic, we should change tounicode as well..

But both `bytes` and `bytearray` provide `.decode()`.